src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -58,15 +58,17 @@ public interface CourseRepository extends JpaRepository<Course, Long> {
             """)
     List<Course> findAllActive(@Param("now") ZonedDateTime now);
 
-    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments", "exams" })
+    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments" })
     @Query("""
             SELECT DISTINCT c FROM Course c
-            WHERE (c.startDate <= :now
-            	OR c.startDate IS NULL)
-            AND (c.endDate >= :now
-            	OR c.endDate IS NULL)
+            LEFT JOIN FETCH c.exams exams
+            LEFT JOIN FETCH exams.registeredUsers registeredUsers
+            WHERE (c.startDate <= :now OR c.startDate IS NULL)
+            AND (c.endDate >= :now OR c.endDate IS NULL)
+            AND (registeredUsers.id = :#{#userId} OR c.instructorGroupName IN :#{#groupNames} OR exams IS NULL)
+            AND (exams.visibleDate <= :now OR exams IS NULL)
```

Comment:
```suggestion
                AND (c.endDate >= :now OR c.endDate IS NULL)
                AND (registeredUsers.id = :#{#userId} OR c.instructorGroupName IN :#{#groupNames} OR exams IS NULL)
                AND (exams.visibleDate <= :now OR exams IS NULL)
```
IIRC most other repositories indent the AND/OR since they are part of the WHERE clause. It’s not consistent in this class, though.